src/app/component/user/components/profile/shopping-list/shopping-list.component.html

```diff
@@ -1,11 +1,25 @@
-<div class="shopping-list-content">
-    <div class="header-position">
-        <h2 class="header">Shopping list <span>{{ shoppingList.length }}</span></h2>
-        <a class="btn-see-all">See all</a>
-    </div>
-    <ul class="shopping-list">
-        <li class="item" *ngFor="let spoppingItem of getShoppingList()">{{ spoppingItem }}
-            <mat-checkbox class="checkbox" color="primary"></mat-checkbox>
-        </li>
-    </ul>
+<div class="header-position shopping-list-content">
+    <h2 class="header">Shopping list <span>{{ shoppingList?.length }}</span></h2>
+    <a class="btn-see-all">See all</a>
 </div>
+<div class="shopping-list-content" *ngIf="shoppingList; else loading;">
+    <div *ngIf="shoppingList.length; else empty;">
+        <ul class="shopping-list">
+            <li class="item" *ngFor="let shoppingItem of shoppingList">{{ shoppingItem.text }}
+                <mat-checkbox
+                    [checked]='shoppingItem.status === "DONE"' 
+                    class="checkbox" 
+                    color="primary"
+                    (click)="toggleDone(shoppingItem)"
+                ></mat-checkbox>
+            </li>
+        </ul>
+    </div>
+    <ng-template #empty>
+        <p class="noItems">You don't have any items in your Shopping List</p>
+    </ng-template>
+ </div>
+ <ng-template #loading>
+    <mat-spinner color="grey" diameter="70" class="loading"></mat-spinner>
```

Comment:
If you have more than 1 attribute, you should write them in separate rows, so please, rewrite it

Reply:
done